gui.py

```diff
@@ -20,9 +20,30 @@ def __init__(self):
         self.show()
 
     def show_message(self):
-        cardnumber = self.card_number.text()
+        cardnumber = self.card_number.text().replace(" ", "")
         expiration = self.expiry_date.text()
         security_code = self.security_code.text()
+        if len(cardnumber) != 16:
+            if len(cardnumber) == 0:
+                if self.tries < 4:
+                    QtWidgets.QMessageBox.critical(self, "Error", "Oops! Looks like you forgot the credit card number")
+                else:
+                    QtWidgets.QMessageBox.critical(self, "Error", "You forgot your card number dumbass.")
+                self.tries += 1
+                    
+                    
+        if len (cardnumber) > 0:
+            if cardnumber.isdigit():
+                QtWidgets.QMessageBox.critical(self, "Error", f"Oops! Card number must be exactly 16 digits, but yours is {len(cardnumber)} digits long.")
+                return
+            else:
+                if self.tries <4:
+                    QtWidgets.QMessageBox.critical(self, "Error", "Oops! Look like there is a letter in your cardnumber.")
+                    self.tries +=1
+                    return
+                else:
+                    QtWidgets.QMessageBox.critical(self, "Error", "Oops! Look like you are a dumbass, there has NEVER been a letter in a cardnumber in the history of ever.")
+                    return
         
         if cardnumber and expiration and security_code:
```

Comment:
Your added logic needs to go under this

Reply:
FUCK YOU

Reply:
UNDER FUCKING WHERE

Reply:
line 48

Reply:
well the line number change when I move code  around so that isnt helpful